private-templates-service/client/src/components/Designer/SaveModal/SaveModal.tsx

```diff
@@ -0,0 +1,149 @@
+    templateName: state.currentTemplate.templateName,
+    sampleDataJSON: state.currentTemplate.sampleDataJSON,
+    version: state.currentTemplate.version
+  }
+}
+
+interface Props {
+  templateID?: string;
+  templateName?: string;
+  sampleDataJSON?: object;
+  templateJSON?: object;
+  designerSampleData?: any;
+  designerTemplateJSON?: any;
+  version?: string;
+  closeModal: () => void;
+  updateTemplate: (templateID?: string, currentVersion?: string, templateJSON?: object, sampleDataJSON?: object, templateName?: string, state?: PostedTemplate.StateEnum, tags?: string[], isShareable?: boolean ) => any; 
+}
+
+interface State {
+  tags: string[];
+  templateName: string;
+}
+
+const mapDispatchToProps = (dispatch: any) => {
+  return { 
+    closeModal: () => {
+      dispatch(closeModal());
+    },
+    updateTemplate: (templateID?: string, currentVersion?: string, templateJSON?: object, sampleDataJSON?: object, templateName?: string, templateState?: PostedTemplate.StateEnum, templateTags?:string[]) => {
+      dispatch(updateTemplate(templateID, currentVersion, templateJSON, sampleDataJSON, templateName, templateState, templateTags));
+    } 
+  }
+}
+
+class SaveModal extends React.Component<Props,State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {tags : [], templateName: "Untitled Template"}
+  }
+
+  saveTags = (tagsToUpdate: string[]) => {
+    this.setState({tags: tagsToUpdate});
+  }
+
+  tagRemove = (tag: string) => {
+    const newTags = this.state.tags.filter((existingTag: string) => existingTag !== tag);
+    this.setState({tags: newTags});
+  }
+
+  onChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+    if(newValue){
+      this.setState({templateName: newValue});
+    }
+  }
+
+  onClick = () => {
+    // will only trigger on first save
+    if(JSON.stringify(this.props.sampleDataJSON) !== JSON.stringify(this.props.designerTemplateJSON) || this.props.sampleDataJSON !== this.props.designerSampleData ){
+      this.props.updateTemplate(undefined,undefined,this.props.designerTemplateJSON,this.props.designerSampleData,this.state.templateName,PostedTemplate.StateEnum.Draft,this.state.tags);
+    }
+    this.props.closeModal();
+  }
+  
+  render(){ 
+    var adaptiveCard = new AdaptiveCards.AdaptiveCard();
+    adaptiveCard.hostConfig = new AdaptiveCards.HostConfig({
+      fontFamily: "Segoe UI, Helvetica Neue, sans-serif"
+    });
+    adaptiveCard.parse(this.props.designerTemplateJSON);
+    var renderedCard = adaptiveCard.render();
+    
+    return(
+      <BackDrop>
+        <Modal>
+          <ColumnWrapper>
+            <TitleWrapper>
+              <StyledTitle> Save Card </StyledTitle>
+              <div>Your card will be saved as a draft until you publish it to your organization.</div>
+            </TitleWrapper>
+            <MiddleRowWrapper>
+              <Container style={{marginRight: "0px", marginLeft:"42px"}} >
```

Comment:
Could styling be moved into styled.tsx file?

Reply:
Styling is inherited from another component, by using Inline I am able to overwrite CSS without changing the component file